juju/juju
# ec2 provider reacts to invalid credential.

## Description of change

This is the first of the providers to learn to react to invalid credential.

Essentially a successfully bootstrapped system might get the cloud credential it uses invalidated by the provider, say cloud credential gets expired... This event should pause all the interactions with the cloud api until the credential has been updated.

This PR ensures that this happens for aws.

## QA steps

* bootstrap
* add models, deploy applications
* invalidate cloud credential
* check that cannot deploy anything nor create anything that needs to also exist on the cloud
* check logs do not contain infinite number of "invalid credential" errors
* check that all workers that require valid credentials are stopped
* update credential or re-validate it
* check that everything is back to fully functional
